src/stream/src/executor/aggregation/mod.rs

```diff
@@ -90,19 +92,17 @@ pub fn agg_call_filter_res(
     }
 }
 
-/// State table and column mapping for `MaterializedState` variant of `AggCallState`.
-pub struct AggStateTable<S: StateStore> {
-    pub table: StateTable<S>,
-    pub mapping: StateTableColumnMapping,
-}
-
-pub fn for_each_agg_state_table<S: StateStore, F: Fn(&mut AggStateTable<S>)>(
-    agg_state_tables: &mut [Option<AggStateTable<S>>],
-    f: F,
-) {
-    agg_state_tables
+pub fn for_each_table_storage<S, F>(state_storages: &mut [AggStateStorage<S>], f: F)
+where
+    S: StateStore,
+    F: Fn(&mut StateTable<S>),
+{
+    state_storages
         .iter_mut()
-        .filter_map(Option::as_mut)
+        .filter_map(|storage| match storage {
+            AggStateStorage::ResultValue => None,
+            AggStateStorage::MaterializedInput { table, .. } => Some(table),
+        })
```

Comment:
How about extracting the filter-mapped iterator without `for_each`, naming `iter_table`? So that more patterns can reuse.

Reply:
The logic here is to do the same thing on all state tables in all `AggStateStorage`, typically `commit` and `init_epoch`. If later we have other variant that also has a state table and maybe other metadata similar to `column_mapping`, we likely need to do `commit` and `init_epoch` on it as well.

Reply:
Yes. I got it